diff --git a/node_modules/next/dist/build/webpack/plugins/flight-client-entry-plugin.js b/node_modules/next/dist/build/webpack/plugins/flight-client-entry-plugin.js
index 0840474..6d3a5c0 100644
--- a/node_modules/next/dist/build/webpack/plugins/flight-client-entry-plugin.js
+++ b/node_modules/next/dist/build/webpack/plugins/flight-client-entry-plugin.js
@@ -126,7 +126,7 @@ class FlightClientEntryPlugin {
                 const modResource = modPath ? modPath.startsWith(_constants1.BARREL_OPTIMIZATION_PREFIX) ? (0, _utils1.formatBarrelOptimizedResource)(mod.resource, modPath) : modPath + modQuery : mod.resource;
                 if (typeof modId !== 'undefined' && modResource) {
                     if (mod.layer === _constants.WEBPACK_LAYERS.reactServerComponents) {
-                        const key = _path.default.relative(compiler.context, modResource).replace(/\/next\/dist\/esm\//, '/next/dist/');
+                        const key = _path.default.relative(compiler.context, modResource).replace(/\\/g, '/').replace(/\/next\/dist\/esm\//, '/next/dist/');
                         const moduleInfo = {
                             moduleId: modId,
                             async: compilation.moduleGraph.isAsync(mod)
@@ -149,7 +149,7 @@ class FlightClientEntryPlugin {
                     let ssrNamedModuleId = _path.default.relative(compiler.context, modResource);
                     if (!ssrNamedModuleId.startsWith('.')) {
                         // TODO use getModuleId instead
-                        ssrNamedModuleId = `./${(0, _normalizepathsep.normalizePathSep)(ssrNamedModuleId)}`;
+                        ssrNamedModuleId = `./${ssrNamedModuleId.replace(/\\/g, '/')}`;
                     }
                     const moduleInfo = {
                         moduleId: modId,
@@ -540,10 +540,11 @@ class FlightClientEntryPlugin {
         // server is using the ESM build (when using the Edge runtime), we need to
         // replace them.
         const clientBrowserLoader = `next-flight-client-entry-loader?${(0, _querystring.stringify)({
-            modules: (this.isEdgeServer ? modules.map(({ request, ids })=>({
-                    request: request.replace(/[\\/]next[\\/]dist[\\/]esm[\\/]/, '/next/dist/'.replace(/\//g, _path.default.sep)),
-                    ids
-                })) : modules).map((x)=>JSON.stringify(x)),
+            modules: modules.map(({ request, ids })=>{
+                let normalizedRequest = request.replace(/\\/g, '/');
+                if (this.isEdgeServer) normalizedRequest = normalizedRequest.replace(/\/next\/dist\/esm\//, '/next/dist/');
+                return { request: normalizedRequest, ids };
+            }).map((x)=>JSON.stringify(x)),
             server: false
         })}!`;
         const clientServerLoader = `next-flight-client-entry-loader?${(0, _querystring.stringify)({
diff --git a/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js b/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js
index 3d300cb..1cd688c 100644
--- a/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js
+++ b/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js
@@ -106,6 +106,7 @@ function getAppPathRequiredChunks(chunkGroup, excludedFiles) {
 // - app/[...catchAll]/page -> app
 // - app/foo/@slot/[...catchAll]/page -> app/foo
 function entryNameToGroupName(entryName) {
+    entryName = entryName.replace(/\\/g, '/');
     let groupName = entryName.slice(0, entryName.lastIndexOf('/'))// Remove slots
     .replace(/\/@[^/]+/g, '')// Remove the group with lookahead to make sure it's not interception route
     .replace(/\/\([^/]+\)(?=(\/|$))/g, '')// Remove catch-all routes since they should be part of the parent group that the catch-all would apply to.
@@ -176,6 +177,7 @@ class ClientReferenceManifestPlugin {
             } else if (!/^app[\\/]/.test(entryName)) {
                 continue;
             }
+            entryName = entryName.replace(/\\/g, '/');
             const manifest = {
                 moduleLoading: {
                     prefix,
@@ -189,7 +191,7 @@ class ClientReferenceManifestPlugin {
                 edgeRscModuleMapping: {}
             };
             // Absolute path without the extension
-            const chunkEntryName = (this.appDirBase + entryName).replace(/[\\/]/g, _path.default.sep);
+            const chunkEntryName = (this.appDirBase + entryName).replace(/\\/g, '/');
             manifest.entryCSSFiles[chunkEntryName] = entrypoint.getFiles().filter((f)=>!f.startsWith('static/css/pages/') && f.endsWith('.css')).map((file)=>{
                 const source = compilation.getAsset(file).source.source();
                 if (this.experimentalInlineCss && // Inline CSS currently does not work properly with HMR, so we only
@@ -209,7 +211,7 @@ class ClientReferenceManifestPlugin {
             const requiredChunks = getAppPathRequiredChunks(entrypoint, rootMainFiles);
             const recordModule = (modId, mod)=>{
                 var _mod_resourceResolveData, _mod_resourceResolveData1, _mod_matchResource;
-                let resource = mod.type === 'css/mini-extract' ? mod.identifier().slice(mod.identifier().lastIndexOf('!') + 1) : mod.resource;
+                let resource = (mod.type === 'css/mini-extract' ? mod.identifier().slice(mod.identifier().lastIndexOf('!') + 1) : mod.resource).replace(/\\/g, '/');
                 if (!resource) {
                     return;
                 }
@@ -221,12 +223,13 @@ class ClientReferenceManifestPlugin {
                 // Note that this isn't that reliable as webpack is still possible to assign
                 // additional queries to make sure there's no conflict even using the `named`
                 // module ID strategy.
-                let ssrNamedModuleId = (0, _path.relative)(context, ((_mod_resourceResolveData = mod.resourceResolveData) == null ? void 0 : _mod_resourceResolveData.path) || resource);
-                const rscNamedModuleId = (0, _path.relative)(context, ((_mod_resourceResolveData1 = mod.resourceResolveData) == null ? void 0 : _mod_resourceResolveData1.path) || resource);
-                if (!ssrNamedModuleId.startsWith('.')) ssrNamedModuleId = `./${ssrNamedModuleId.replace(/\\/g, '/')}`;
+                let ssrNamedModuleId = _path.default.relative(context, ((_mod_resourceResolveData = mod.resourceResolveData) == null ? void 0 : _path.default.resolve(_mod_resourceResolveData.path)) || resource).replace(/\\/g, '/');
+                let rscNamedModuleId = _path.default.relative(context, ((_mod_resourceResolveData1 = mod.resourceResolveData) == null ? void 0 : _path.default.resolve(_mod_resourceResolveData1.path)) || resource).replace(/\\/g, '/');
+                if (!ssrNamedModuleId.startsWith('.')) ssrNamedModuleId = `./${ssrNamedModuleId}`;
+                if (!rscNamedModuleId.startsWith('.')) rscNamedModuleId = `./${rscNamedModuleId}`;
                 // The client compiler will always use the CJS Next.js build, so here we
                 // also add the mapping for the ESM build (Edge runtime) to consume.
-                const esmResource = /[\\/]next[\\/]dist[\\/]/.test(resource) ? resource.replace(/[\\/]next[\\/]dist[\\/]/, '/next/dist/esm/'.replace(/\//g, _path.default.sep)) : null;
+                const esmResource = /[\\/]next[\\/]dist[\\/]/.test(resource) ? resource.replace(/[\\/]next[\\/]dist[\\/]/, '/next/dist/esm/').replace(/\\/g, '/') : null;
                 // An extra query param is added to the resource key when it's optimized
                 // through the Barrel Loader. That's because the same file might be created
                 // as multiple modules (depending on what you import from it).
@@ -423,7 +426,7 @@ class ClientReferenceManifestPlugin {
                 }
             }
             const json = JSON.stringify(mergedManifest);
-            const pagePath = pageName.replace(/%5F/g, '_');
+            const pagePath = pageName.replace(/\\/g, '/').replace(/%5F/g, '_');
             const pageBundlePath = (0, _normalizepagepath.normalizePagePath)(pagePath.slice('app'.length));
             compilation.emitAsset('server/app' + pageBundlePath + '_' + _constants.CLIENT_REFERENCE_MANIFEST + '.js', new _webpack.sources.RawSource(`globalThis.__RSC_MANIFEST=(globalThis.__RSC_MANIFEST||{});globalThis.__RSC_MANIFEST[${JSON.stringify(pagePath.slice('app'.length))}]=${json};`));
         }
diff --git a/node_modules/next/dist/esm/build/webpack/plugins/flight-client-entry-plugin.js b/node_modules/next/dist/esm/build/webpack/plugins/flight-client-entry-plugin.js
index 35395e9..9d352de 100644
--- a/node_modules/next/dist/esm/build/webpack/plugins/flight-client-entry-plugin.js
+++ b/node_modules/next/dist/esm/build/webpack/plugins/flight-client-entry-plugin.js
@@ -112,7 +112,7 @@ export class FlightClientEntryPlugin {
                 const modResource = modPath ? modPath.startsWith(BARREL_OPTIMIZATION_PREFIX) ? formatBarrelOptimizedResource(mod.resource, modPath) : modPath + modQuery : mod.resource;
                 if (typeof modId !== 'undefined' && modResource) {
                     if (mod.layer === WEBPACK_LAYERS.reactServerComponents) {
-                        const key = path.relative(compiler.context, modResource).replace(/\/next\/dist\/esm\//, '/next/dist/');
+                        const key = path.relative(compiler.context, modResource).replace(/\\/g, '/').replace(/\/next\/dist\/esm\//, '/next/dist/');
                         const moduleInfo = {
                             moduleId: modId,
                             async: compilation.moduleGraph.isAsync(mod)
@@ -135,7 +135,7 @@ export class FlightClientEntryPlugin {
                     let ssrNamedModuleId = path.relative(compiler.context, modResource);
                     if (!ssrNamedModuleId.startsWith('.')) {
                         // TODO use getModuleId instead
-                        ssrNamedModuleId = `./${normalizePathSep(ssrNamedModuleId)}`;
+                        ssrNamedModuleId = `./${ssrNamedModuleId.replace(/\\/g, '/')}`;
                     }
                     const moduleInfo = {
                         moduleId: modId,
@@ -526,10 +526,11 @@ export class FlightClientEntryPlugin {
         // server is using the ESM build (when using the Edge runtime), we need to
         // replace them.
         const clientBrowserLoader = `next-flight-client-entry-loader?${stringify({
-            modules: (this.isEdgeServer ? modules.map(({ request, ids })=>({
-                    request: request.replace(/[\\/]next[\\/]dist[\\/]esm[\\/]/, '/next/dist/'.replace(/\//g, path.sep)),
-                    ids
-                })) : modules).map((x)=>JSON.stringify(x)),
+            modules: modules.map(({ request, ids })=>{
+                let normalizedRequest = request.replace(/\\/g, '/');
+                if (this.isEdgeServer) normalizedRequest = normalizedRequest.replace(/\/next\/dist\/esm\//, '/next/dist/');
+                return { request: normalizedRequest, ids };
+            }).map((x)=>JSON.stringify(x)),
             server: false
         })}!`;
         const clientServerLoader = `next-flight-client-entry-loader?${stringify({
diff --git a/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js b/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
index d944daa..7e44471 100644
--- a/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
+++ b/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
@@ -57,6 +57,7 @@ function getAppPathRequiredChunks(chunkGroup, excludedFiles) {
 // - app/[...catchAll]/page -> app
 // - app/foo/@slot/[...catchAll]/page -> app/foo
 function entryNameToGroupName(entryName) {
+    entryName = entryName.replace(/\\/g, '/');
     let groupName = entryName.slice(0, entryName.lastIndexOf('/'))// Remove slots
     .replace(/\/@[^/]+/g, '')// Remove the group with lookahead to make sure it's not interception route
     .replace(/\/\([^/]+\)(?=(\/|$))/g, '')// Remove catch-all routes since they should be part of the parent group that the catch-all would apply to.
@@ -122,6 +123,7 @@ export class ClientReferenceManifestPlugin {
             }
         });
         for (let [entryName, entrypoint] of compilation.entrypoints){
+            entryName = entryName.replace(/\\/g, '/');
             if (entryName === CLIENT_STATIC_FILES_RUNTIME_MAIN_APP || entryName === APP_CLIENT_INTERNALS) {
                 entryName = '';
             } else if (!/^app[\\/]/.test(entryName)) {
@@ -140,7 +142,7 @@ export class ClientReferenceManifestPlugin {
                 edgeRscModuleMapping: {}
             };
             // Absolute path without the extension
-            const chunkEntryName = (this.appDirBase + entryName).replace(/[\\/]/g, path.sep);
+            const chunkEntryName = (this.appDirBase + entryName).replace(/[\\/]/g, '/');
             manifest.entryCSSFiles[chunkEntryName] = entrypoint.getFiles().filter((f)=>!f.startsWith('static/css/pages/') && f.endsWith('.css')).map((file)=>{
                 const source = compilation.getAsset(file).source.source();
                 if (this.experimentalInlineCss && // Inline CSS currently does not work properly with HMR, so we only
@@ -160,7 +162,7 @@ export class ClientReferenceManifestPlugin {
             const requiredChunks = getAppPathRequiredChunks(entrypoint, rootMainFiles);
             const recordModule = (modId, mod)=>{
                 var _mod_resourceResolveData, _mod_resourceResolveData1, _mod_matchResource;
-                let resource = mod.type === 'css/mini-extract' ? mod.identifier().slice(mod.identifier().lastIndexOf('!') + 1) : mod.resource;
+                let resource = (mod.type === 'css/mini-extract' ? mod.identifier().slice(mod.identifier().lastIndexOf('!') + 1) : mod.resource).replace(/\\/g, '/');
                 if (!resource) {
                     return;
                 }
@@ -172,12 +174,13 @@ export class ClientReferenceManifestPlugin {
                 // Note that this isn't that reliable as webpack is still possible to assign
                 // additional queries to make sure there's no conflict even using the `named`
                 // module ID strategy.
-                let ssrNamedModuleId = relative(context, ((_mod_resourceResolveData = mod.resourceResolveData) == null ? void 0 : _mod_resourceResolveData.path) || resource);
-                const rscNamedModuleId = relative(context, ((_mod_resourceResolveData1 = mod.resourceResolveData) == null ? void 0 : _mod_resourceResolveData1.path) || resource);
-                if (!ssrNamedModuleId.startsWith('.')) ssrNamedModuleId = `./${ssrNamedModuleId.replace(/\\/g, '/')}`;
+                let ssrNamedModuleId = relative(context, ((_mod_resourceResolveData = mod.resourceResolveData) == null ? void 0 : _mod_resourceResolveData.path) || resource).replace(/\\/g, '/');
+                let rscNamedModuleId = relative(context, ((_mod_resourceResolveData1 = mod.resourceResolveData) == null ? void 0 : _mod_resourceResolveData1.path) || resource).replace(/\\/g, '/');
+                if (!ssrNamedModuleId.startsWith('.')) ssrNamedModuleId = `./${ssrNamedModuleId}`;
+                if (!rscNamedModuleId.startsWith('.')) rscNamedModuleId = `./${rscNamedModuleId}`;
                 // The client compiler will always use the CJS Next.js build, so here we
                 // also add the mapping for the ESM build (Edge runtime) to consume.
-                const esmResource = /[\\/]next[\\/]dist[\\/]/.test(resource) ? resource.replace(/[\\/]next[\\/]dist[\\/]/, '/next/dist/esm/'.replace(/\//g, path.sep)) : null;
+                const esmResource = /[\\/]next[\\/]dist[\\/]/.test(resource) ? resource.replace(/[\\/]next[\\/]dist[\\/]/, '/next/dist/esm/') : null;
                 // An extra query param is added to the resource key when it's optimized
                 // through the Barrel Loader. That's because the same file might be created
                 // as multiple modules (depending on what you import from it).
