diff --git a/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js b/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js
index 3d300cb..3ad6bb7 100644
--- a/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js
+++ b/node_modules/next/dist/build/webpack/plugins/flight-manifest-plugin.js
@@ -106,6 +106,7 @@ function getAppPathRequiredChunks(chunkGroup, excludedFiles) {
 // - app/[...catchAll]/page -> app
 // - app/foo/@slot/[...catchAll]/page -> app/foo
 function entryNameToGroupName(entryName) {
+    entryName = entryName.replace(/\\/g, '/');
     let groupName = entryName.slice(0, entryName.lastIndexOf('/'))// Remove slots
     .replace(/\/@[^/]+/g, '')// Remove the group with lookahead to make sure it's not interception route
     .replace(/\/\([^/]+\)(?=(\/|$))/g, '')// Remove catch-all routes since they should be part of the parent group that the catch-all would apply to.
@@ -189,7 +190,7 @@ class ClientReferenceManifestPlugin {
                 edgeRscModuleMapping: {}
             };
             // Absolute path without the extension
-            const chunkEntryName = (this.appDirBase + entryName).replace(/[\\/]/g, _path.default.sep);
+            const chunkEntryName = (this.appDirBase + entryName).replace(/[\\/]/g, '/');
             manifest.entryCSSFiles[chunkEntryName] = entrypoint.getFiles().filter((f)=>!f.startsWith('static/css/pages/') && f.endsWith('.css')).map((file)=>{
                 const source = compilation.getAsset(file).source.source();
                 if (this.experimentalInlineCss && // Inline CSS currently does not work properly with HMR, so we only
diff --git a/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js b/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
index d944daa..69fc400 100644
--- a/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
+++ b/node_modules/next/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
@@ -57,6 +57,7 @@ function getAppPathRequiredChunks(chunkGroup, excludedFiles) {
 // - app/[...catchAll]/page -> app
 // - app/foo/@slot/[...catchAll]/page -> app/foo
 function entryNameToGroupName(entryName) {
+    entryName = entryName.replace(/\\/g, '/');
     let groupName = entryName.slice(0, entryName.lastIndexOf('/'))// Remove slots
     .replace(/\/@[^/]+/g, '')// Remove the group with lookahead to make sure it's not interception route
     .replace(/\/\([^/]+\)(?=(\/|$))/g, '')// Remove catch-all routes since they should be part of the parent group that the catch-all would apply to.
@@ -140,7 +141,7 @@ export class ClientReferenceManifestPlugin {
                 edgeRscModuleMapping: {}
             };
             // Absolute path without the extension
-            const chunkEntryName = (this.appDirBase + entryName).replace(/[\\/]/g, path.sep);
+            const chunkEntryName = (this.appDirBase + entryName).replace(/[\\/]/g, '/');
             manifest.entryCSSFiles[chunkEntryName] = entrypoint.getFiles().filter((f)=>!f.startsWith('static/css/pages/') && f.endsWith('.css')).map((file)=>{
                 const source = compilation.getAsset(file).source.source();
                 if (this.experimentalInlineCss && // Inline CSS currently does not work properly with HMR, so we only
